src/Command/DrupalUpdate/UpdateScriptUtility.php

```diff
@@ -0,0 +1,317 @@
+            echo $v . str_repeat(" ", 1) . " | ";
+          }
+
+        }
+        $i++;
+      }
+      echo "\n";
+    }
+    echo "\n";
+  }
+
+  /**
+   * Update code based on available security update.
+   * @param $latest_security_updates
+   */
+
+  function updateCode($latest_security_updates) {
+    foreach ($latest_security_updates as $k => $value){
+      if(!isset($value['download_link'])){
+        continue;
+      }
+      if($value['package']=='drupal'){
+        $dirname = 'temp_drupal_core';
+        $filename = $value['file_path'] . "/" . $dirname . "";
+        if (!file_exists($filename)) {
+          $oldmask = umask(0);
+          mkdir($value['file_path'] . "/" . $dirname, 0777);
+          umask($oldmask);
+          $value['file_path'] = $value['file_path'] . "/" . $dirname . "";
+        } else {
+          echo "The directory $dirname exists.";
+        }
+      }
+      if(is_array($value['file_path'])){
+        foreach ($value['file_path'] as $item) {
+          $this->download_remote_file($value['package'], $value['download_link'], $item);
+        }
+      }else{
+        $this->download_remote_file($value['package'], $value['download_link'], $value['file_path']);
+      }
+
+    }
+
+  }
+
+  /**
+   * Download and extract tar files in given directory path.
+   * @param $package
+   * @param $file_url
+   * @param $save_to
+   */
+  function download_remote_file($package, $file_url, $save_to) {
+    if($package == 'drupal'){
+      $this->download_remote_file_drupal_core($package, $file_url, $save_to);
+      return;
+    }
+    $content = file_get_contents($file_url);
+    file_put_contents($save_to . '/' . $package . '.tar.gz', $content);
+    try {
+      $phar = new PharData($save_to . '/' . $package . '.tar.gz');
+      $this->rrmdir($save_to . '/' . $package);
+      $phar->extractTo($save_to, NULL, TRUE); // extract all files
+    } catch (\Exception $e) {
+      // handle errors
+    }
+  }
+
+  /**
+   * Download and extract tar files in drupal core temp directory path.
+   * @param $package
+   * @param $file_url
+   * @param $save_to
+   */
+  function download_remote_file_drupal_core($package, $file_url, $save_to) {
+    $content = file_get_contents($file_url);
+    $folder_name = str_replace('.tar.gz', '', basename($file_url));
+    file_put_contents($save_to . '/' . $package . '.tar.gz', $content);
+    try {
+      $phar = new PharData($save_to . '/' . $package . '.tar.gz');
+      $this->rrmdir($save_to . '/' . $package);
+      $phar->extractTo($save_to, NULL, TRUE); // extract all files
+      rename($save_to . '/' . $folder_name, $save_to . '/drupal');
+    } catch (\Exception $e) {
+      // handle errors
+    }
+    if($package == 'drupal'){
+      // Replace the folder to inside docroot folder.
+      $this->coreUpdate($save_to . '/drupal');
+      $this->rrmdir($save_to);
+    }
+  }
+
+  /**
+   * Remove directory and sub directory.
+   * @param $dir
+   */
+  function rrmdir($dir) {
```

Comment:
My suggestion - create a separate utility or use some symfony package to deal with files and folders.  Check https://symfony.com/doc/current/components/filesystem.html

Reply:
Definitely use Symfony.